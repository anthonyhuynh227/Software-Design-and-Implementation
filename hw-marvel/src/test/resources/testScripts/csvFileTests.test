#Test for a graph build on csv file with load graph

#create graph
LoadGraph CS-Student csStudent.csv

#Print all nodes in graph
ListNodes CS-Student

#Print all child node with outgoing edge
ListChildren CS-Student Jasmine

#Test one node is not in graph
FindPath CS-Student John Brian

#Test two node are not in graph
FindPath CS-Student Brian Lulu

#Test path is not exist in graph
FindPath CS-Student Jack Jackie

#Test path from a node to itself
FindPath CS-Student Jack Jack

#Test shortest path
FindPath CS-Student John Tom

#Test shortest path with lexicographically path
FindPath CS-Student John Jasmine