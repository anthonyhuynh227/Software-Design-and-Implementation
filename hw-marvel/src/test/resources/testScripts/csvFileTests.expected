#Test for a graph build on csv file with load graph

#create graph
loaded graph CS-Student

#Print all nodes in graph
CS-Student contains: Anthony Jack Jackie Jasmine John Tom

#Print all child node with outgoing edge
the children of Jasmine in CS-Student are: Anthony(CSE332) Jack(CSE332) Tom(CSE421)

#Test one node is not in graph
unknown: Brian

#Test two node are not in graph
unknown: Brian
unknown: Lulu

#Test path is not exist in graph
path from Jack to Jackie:
no path found

#Test path from a node to itself
path from Jack to Jack:

#Test shortest path
path from John to Tom:
John to Jack via CSE253
Jack to Tom via CSE550

#Test shortest path with lexicographically path
path from John to Jasmine:
John to Anthony via CSE150
Anthony to Jasmine via CSE332
